# Reproducible Data Workflow for Video Game Sales, Ratings, and Genre Trends

**Student:** Robert Mayfield  
**Project:** AI Programming Foundations - Udacity AI MBA Capstone  
**Dataset:** Video Game Sales, Ratings, and Genre Data

## 1. Project Overview

This project focuses on building a clean, reusable, and reproducible data workflow using Python, Pandas, NumPy, Matplotlib, and Seaborn. The `Video Game Sales and Ratings` dataset contains video game market information such as game titles, platforms, genres, publishers, sales figures, ratings, and review scores. The goal is to clean and explore the dataset, create meaningful visualizations, and summarize insights that help establish industry context for a future AI-assisted game development system called AI Game Director Studio.

### Core Question

What patterns in video game genre, platform, sales, and ratings data can inform the desgin focus of an AI-assisted game development tool?

## 2. Setup

The following imports support data loading and initial inspection. Additional libraries will be added in later sections as needed.

In [4]:
import sys
import os
sys.path.insert(0, os.path.abspath('..'))

import zipfile
import pandas as pd

from src.inspection import inspect_dataframe

## 3. Data Ingestion

The dataset is stored as a zip archive at `data/raw/video-game-sales-ratings-data.zip` and contains a single CSV file, `Video_Games.csv`. The data is loaded directly from the archive without extracting to disk, keeping the raw data directory clean and the workflow self contained.

In [5]:
ZIP_PATH = '../data/raw/video-game-sales-ratings-data.zip'
CSV_NAME = 'Video_Games.csv'

with zipfile.ZipFile(ZIP_PATH, 'r') as z:
    with z.open(CSV_NAME) as f:
        df = pd.read_csv(f, index_col=0)

df.head()

,Name,Platform,Year_of_Release,Genre,Publisher,NA_Sales,EU_Sales,JP_Sales,Other_Sales,Global_Sales,Critic_Score,Critic_Count,User_Score,User_Count,Developer,Rating
index,,,,,,,,,,,,,,,,
0,Wii Sports,Wii,2006.0,Sports,Nintendo,41.36,28.96,3.77,8.45,82.53,76.0,51.0,8,322.0,Nintendo,E
1,Super Mario Bros.,NES,1985.0,Platform,Nintendo,29.08,3.58,6.81,0.77,40.24,NaN,NaN,NaN,NaN,NaN,NaN
2,Mario Kart Wii,Wii,2008.0,Racing,Nintendo,15.68,12.76,3.79,3.29,35.52,82.0,73.0,8.3,709.0,Nintendo,E
3,Wii Sports Resort,Wii,2009.0,Sports,Nintendo,15.61,10.93,3.28,2.95,32.77,80.0,73.0,8,192.0,Nintendo,E
4,Pokemon Red/Pokemon Blue,GB,1996.0,Role-Playing,Nintendo,11.27,8.89,10.22,1.00,31.37,NaN,NaN,NaN,NaN,NaN,NaN


## 4. Initial Dataset Inspection

Before cleaning or analysis, we run a full inspection of the dataset to understand its structure, identify missing values, spot duplicates, and get a sense of the distributions across numeric and categorical columns. The `inspect_dataframe` function from `src/inspection.py` packages these steps into a single reusable call.

In [6]:
inspect_dataframe(df)

SHAPE AND STRUCTURE
------------------------------------------------------------
Rows:    16,928
Columns: 16

COLUMN NAMES AND DATA TYPES
------------------------------------------------------------
Name                object
Platform            object
Year_of_Release    float64
Genre               object
Publisher           object
NA_Sales           float64
EU_Sales           float64
JP_Sales           float64
Other_Sales        float64
Global_Sales       float64
Critic_Score       float64
Critic_Count       float64
User_Score          object
User_Count         float64
Developer           object
Rating              object

MEMORY USAGE
------------------------------------------------------------
Total: 8.19 MB

MISSING VALUES
------------------------------------------------------------
                 Null Count  Null %
User_Count             9210   54.41
Critic_Count           8668   51.21
Critic_Score           8668   51.21
Rating                 6836   40.38
User_Score            

### Inspection Notes

**Shape and Structure:** The dataset contains 16,928 rows and 16 columns. Columns cover game titles, platforms, release years, genres, publishers, four regional sales figures, global sales, critic and user review scores, developer names, and ESRB ratings. Total memory footprint is 8.19 MB.

**Data Types:** All sales and score columns load as float64 except `User_Score`, which loads as object. The categorical describe output reveals why: the most frequent value in `User_Score` is `tbd`, a placeholder used when a user score is not yet available. This string presence prevents pandas from inferring a numeric type. `Year_of_Release` also loads as float64 rather than integer, likely due to missing values in that column. Both will be addressed during cleaning.

**Missing Values:** The dataset has substantial missingness concentrated in review related columns. `User_Count` is missing for 9,210 records (54.41%), `Critic_Count` and `Critic_Score` are each missing for 8,668 records (51.21%), and `Rating`, `User_Score`, and `Developer` are missing for roughly 40% of records each. This pattern is consistent across the dataset: older and lower profile titles were frequently not reviewed by critics or users. `Year_of_Release` is missing for 273 records (1.61%), `Publisher` for 55 (0.32%), and `Name` and `Genre` for just 2 records each (0.01%). The review column gaps are significant enough that any analysis involving scores will be working with roughly half the dataset.

**Duplicate Rows:** 209 duplicate rows are present and will be removed during cleaning.

**Numeric Distributions:** Sales figures are heavily right skewed across all regions. The median global sales figure is 0.17 million units while the maximum is 82.53 million, and the 75th percentile sits at just 0.49 million. The vast majority of titles sell modestly, with a small number of blockbusters driving the upper range. `Critic_Score` ranges from 13 to 98 with a mean of 69, suggesting most reviewed titles cluster in the average to good range. `User_Count` has an extreme distribution with a mean of 163 but a maximum of 10,665 and a large standard deviation, indicating that a handful of titles attract enormous user review volume while most receive very few. Release years span 1980 to 2020 with a mean of 2006.5, reflecting that the bulk of the dataset covers the mid-2000s peak of the disc-based console era.

**Categorical Summaries:** There are 31 unique platforms in the dataset, with PS2 being the most represented at 2,188 titles. Genre has only 12 unique values, with Action being the most common at 3,410 titles. Publisher has 581 unique values, with Electronic Arts appearing most frequently at 1,389 titles. Developer has 1,696 unique values. Rating has 8 unique values with E (Everyone) being the most common at 4,043 titles, followed by other ESRB categories.

**Sample Data:** The first rows confirm the dataset is sorted by global sales descending, with Nintendo titles dominating the top. The last rows show titles clustered around 0.6 million in global sales, and several are missing review data entirely, consistent with the missing value patterns described above.

## 5. Data Cleaning Functions

## 6. Exploratory Analysis Function

## 7. Visualizations

## 8. Summary and Interpretation

## 9. Responsible Data Handling and Bias Awareness

## 10. Future Integration with AI Game Director Studio